core/src/main/scala/org/http4s/Query.scala

```diff
@@ -1,13 +1,82 @@
 package org.http4s
 
-import org.http4s.Query._
+import org.http4s.FormQuery._
 import org.http4s.parser.QueryParser
-import org.http4s.util.{UrlFormCodec, UrlCodingUtils, Writer, Renderable}
+import org.http4s.util._
+import org.http4s.util.encoding.UriCodingUtils
 
 import scala.collection.generic.CanBuildFrom
-import scala.collection.immutable.{BitSet, IndexedSeq}
+import scala.collection.immutable.IndexedSeq
 import scala.collection.mutable.ListBuffer
-import scala.collection.{ IndexedSeqOptimized, mutable }
+import scala.collection.{IndexedSeqOptimized, mutable}
+
+
+sealed trait Query extends QueryOps with Renderable {
+  def asForm: FormQuery
+  def isEmpty: Boolean
+  def encoded: String
+}
+
+/** Represents the absence of a query component */
+object EmptyQuery extends Query {
```

Comment:
Yes, if that's `/path` (no query) is what `Query.empty` is supposed to signify, I'm not sure I was totally sure.